ambientweather2sqlite/database.py

```
@@ -178,140 +92,268 @@ def _select_parts_from_aggregation_fields(
 
     return select_parts
 
+class DatabaseManager:
+    """Manages persistent database connections and operations."""
+    
+    def __init__(self, db_path: str):
+        if not Path(db_path).exists():
+            self._create_database_if_not_exists(db_path)
+        self.conn: sqlite3.Connection = sqlite3.connect(db_path, check_same_thread=False)
```

Ruff (0.11.9)

Line too long (89 > 88)

(E501)